lib/stimpack/integrations/rails.rb

```diff
@@ -1,17 +1,53 @@
 # frozen_string_literal: true
 
+require "active_support/inflections"
+
 module Stimpack
   module Integrations
     class Rails
-      def self.install(app)
+      def initialize(app)
+        @app = app
+
         Stimpack.config.paths.freeze
+        create_engines
+        inject_paths
+      end
+
+      def create_engines
+        Packs.all.each do |pack|
+          next unless pack.config.engine?
+
+          pack.engine = create_engine(pack)
+        end
+      end
 
-        Packs.each do |pack|
+      def inject_paths
+        Packs.all.each do |pack|
           Stimpack.config.paths.each do |path|
-            app.paths[path] << pack.path.join(path)
+            @app.paths[path] << pack.path.join(path)
           end
         end
       end
+
+      private
+
+      def create_namespace(name)
```

Comment:
The implementation of these hasn't changed right? Just the location of the methods?